test/notifications/display_test.dart

```diff
@@ -89,6 +89,28 @@ RemoveFcmMessage removeFcmMessage(List<Message> zulipMessages, {Account? account
   }) as RemoveFcmMessage;
 }
 
+Uri messageIntentDataUrl(MessageFcmMessage data) {
```

Comment:
I think this can be replaced with just calling `NotificationOpenPayload.buildUrl`. We'll want unit tests for that class as I mentioned above; but those will give us test coverage on that method more thoroughly anyway, I think, than we get from reproducing the URLs here in the context of test cases that are focused on other logic (namely `_onMessageFcmMessage`).